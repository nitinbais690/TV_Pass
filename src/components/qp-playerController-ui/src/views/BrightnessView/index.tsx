import useAppColors from 'core/presentation/hooks/use-app-colors';
import React, { useEffect, useState } from 'react';
import { styles } from '../BrightnessView/styles';
import { appDimensionValues } from 'core/styles/AppStyles';
import LinearGradient from 'react-native-linear-gradient';
import Brightnessicon from 'assets/images/sun.svg';
import { Text } from 'react-native';
import Slider from '@react-native-community/slider';
import SystemSetting from 'react-native-system-setting';

export default function BrightnessView(props: BlackBgWithBorderProps) {
    const appColors = useAppColors();
    const style = styles(appColors);
    const [brightness, setBrightness] = useState<number>(0);

    useEffect(() => {
        if (brightness == 0) {
            SystemSetting.setAppBrightness(props.initialBrightness / 100);
            setBrightness(props.initialBrightness);
        }
    }, [brightness, props.initialBrightness]);

    return (
        <LinearGradient
            colors={['rgba(255, 255, 255, 0.2)', 'rgba(255, 255, 255, 0)']}
            useAngle={true}
            angle={90}
            style={[style.containerStyle, props.style]}>
            <Text style={style.plusTextStyle}>+</Text>
            <Slider
                maximumValue={100}
                minimumValue={0}
                value={brightness}
                step={1}
                style={style.sliderStyle}
                onSlidingComplete={(value: number) => {
                    setBrightness(value);
                    SystemSetting.setAppBrightness(value / 100);
                    props.onCompleteBrightness(value);
                }}
                onValueChange={(value: number) => {
                    setBrightness(value);
                    SystemSetting.setAppBrightness(value / 100);
                }}
                thumbTintColor={appColors.secondary}
                minimumTrackTintColor={appColors.secondary}
                maximumTrackTintColor={appColors.secondary}
            />
            <Text style={style.minusTextStyle}>-</Text>
            <Brightnessicon
                style={style.brightnessIconStyle}
                height={appDimensionValues.xlg}
                width={appDimensionValues.xlg}
            />
        </LinearGradient>
    );
}

interface BlackBgWithBorderProps {
    initialBrightness: number;
    onCompleteBrightness: (brightness: number) => void;
    style?: {};
}
