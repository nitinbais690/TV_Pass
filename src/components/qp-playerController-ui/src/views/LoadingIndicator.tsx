import React from 'react';
import { ActivityIndicator } from 'react-native';
import { loadingIndicatorStyles } from '../styles/PlayerControls.style';

const loadingStyle = loadingIndicatorStyles();
const AppLoadingIndicator = (): JSX.Element => {
    return <ActivityIndicator color={'white'} style={loadingStyle.loading} size="large" />;
};
export default AppLoadingIndicator;
