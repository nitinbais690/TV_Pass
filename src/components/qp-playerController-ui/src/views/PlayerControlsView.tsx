import React, { useReducer, useRef, useState, useEffect, useImperativeHandle } from 'react';
import { Animated, TouchableHighlight, View, Text, Platform } from 'react-native';
import { useDimensions, useLayout } from '@react-native-community/hooks';
import SystemSetting from 'react-native-system-setting';
import {
    playerBottomControlsStyles,
    playerCenterControlsStyles,
    playerTopControlsStyles,
} from '../styles/PlayerControls.style';
import RewindIcon from '../../assets/images/rewind.svg';
import ForwardIcon from '../../assets/images/forward.svg';
import PlayIcon from '../../assets/images/play.svg';
import PauseIcon from '../../assets/images/pause.svg';
import TracksIcon from '../../assets/images/tracks.svg';
import MuteIcon from '../../assets/images/mute.svg';
import UnMuteIcon from '../../assets/images/unmute.svg';
import CloseIcon from '../../assets/images/close.svg';
import { colors } from 'qp-common-ui';
import Slider from '@react-native-community/slider';
import { visualizeVideoDuration } from '../utils/utils';
import { useSafeArea } from 'react-native-safe-area-context';
import TrackSelectionView from './TracksSelectionView';
import { PlaybackStateValue, QpNxgAirplayView } from 'rn-qp-nxg-player';
import { ResourceVm } from 'qp-discovery-ui';
import { TrackInfo } from 'screens/components/PlatformPlayer';

export type CastType = 'Chromecast' | 'Airplay';

export type PlayerControlsActions = {
    show: () => void;
    hide: () => void;
};

export interface PlayControllerProps {
    currentTime: number;
    contentTitle?: string;
    isCasting: boolean;
    isDownloadedContentPlayback?: boolean;
    castType?: CastType;
    castLabel: string;
    playbackDuration: number;
    isLoading?: boolean;
    playbackState?: PlaybackStateValue;
    showOnStart?: boolean;
    showCastIcon?: boolean;
    controlTimeout?: number;
    resource?: ResourceVm | null;
    onBackPress?: () => void;
    onPause?: () => void;
    onPlay?: () => void;
    onStop?: () => void;
    handleVolumeToggle?: (value: number) => void;
    onRewindPress?: () => void;
    onForwardPress?: () => void;
    onAirPlayPress?: () => void;
    onCastPress?: () => void;
    onSlidingComplete?: (value: number) => void;
    captionOptions?: TrackInfo[];
    audioOptions?: TrackInfo[];
    activeTextTrack?: string;
    activeAudioTrack?: string;
    onAudioOptionSelected?: (option: string, languageCode: string) => void;
    onTextOptionSelected?: (option: string, languageCode: string) => void;
}

const playerProps = {
    controlTimeoutDelay: 3000,
    controlTimeout: 0,
};

export const PlayerControlsView = React.forwardRef<PlayerControlsActions, PlayControllerProps>(
    (props, ref): JSX.Element => {
        const isMounted = useRef(true);
        const insets = useSafeArea();
        const { onLayout, ...layout } = useLayout();
        let hideAnimationTimeoutId = useRef<number | undefined>();
        const [showTrackSelection, setShowTrackSelection] = useState(false);
        const { width, height } = useDimensions().window;
        const [sliding, setSliding] = useState(false);
        const [seekedTo, setSeekedTo] = useState<number>(0);
        const [seekVolumeTo, setSeekVolumeTo] = useState<number>(0);
        const [showVolumeIcon, setShowVolumeIcon] = useState(true);

        const isPortrait = height > width;
        const centerControlStyles = playerCenterControlsStyles();
        const topControlStyles = playerTopControlsStyles(insets, isPortrait);
        const bottomControlStyles = playerBottomControlsStyles(insets, isPortrait);

        const initialValue = props.showOnStart ? 1 : 0;

        useEffect(() => {
            isMounted.current = true;

            SystemSetting.getVolume().then(value => {
                setSeekVolumeTo(value * 100);
            });

            let volumeListener = SystemSetting.addVolumeListener(({ value }) => {
                setSeekVolumeTo(value * 100);
            });

            return () => {
                isMounted.current = false;
                if (volumeListener) {
                    SystemSetting.removeVolumeListener(volumeListener);
                }
            };
        }, []);

        useImperativeHandle(ref, () => ({
            show: () => {
                setState({ name: 'showControls', value: true });
                showControlAnimation();
                setState({ name: 'showingOverlay', value: false });
            },
            hide: () => {
                hideControlAnimation(0);
                setState({ name: 'showingOverlay', value: true });
            },
        }));

        const animations = {
            bottomControl: {
                marginBottom: new Animated.Value(0),
                opacity: new Animated.Value(initialValue),
            },
            topControl: {
                marginTop: new Animated.Value(0),
                opacity: new Animated.Value(initialValue),
            },
            centerControl: {
                marginTop: new Animated.Value(0),
                opacity: new Animated.Value(initialValue),
            },
            video: {
                opacity: new Animated.Value(1),
            },
            loader: {
                rotate: new Animated.Value(0),
                MAX_VALUE: 360,
            },
        };

        const initialState = {
            isLoading: props.isLoading,
            currentTime: props.currentTime,
            duration: props.playbackDuration,
            lastScreenPress: 0,
            paused: false,
            showControls: props.showOnStart,
            animations: animations,
            showingOverlay: false,
        };

        const controlsReducer = (state: any, action: any): any => {
            switch (action.name) {
                case 'showControls':
                    return { ...state, showControls: action.value };
                case 'animations':
                    return { ...state, animations: action.value };
                case 'updateState':
                    return { ...state, animations: action.value };
                case 'pauseState':
                    return { ...state, paused: action.value };
                case 'showingOverlay':
                    return { ...state, showingOverlay: action.value };
            }
        };
        const [state, setState] = useReducer(controlsReducer, initialState);

        useEffect(() => {
            if (!isMounted.current) {
                return;
            }
            if (hideAnimationTimeoutId.current && (props.isCasting || props.playbackState === 'PAUSED')) {
                clearTimeout(hideAnimationTimeoutId.current);
            }

            if (state.showingOverlay) {
                // do not show controls when showing an overlay
                return;
            }

            // Always show controls while casting
            if ((props.isCasting || props.playbackState === 'PAUSED') && !state.showControls) {
                showControlAnimation();
                setState({ name: 'showControls', value: true });
            } else {
                hideControlAnimation(0);
            }
            // eslint-disable-next-line react-hooks/exhaustive-deps
        }, [props.isCasting, props.playbackState]);

        useEffect(() => {
            if (props.playbackState === 'PAUSED') {
                setState({ name: 'pauseState', value: true });
            } else if (props.playbackState === 'STARTED') {
                setState({ name: 'pauseState', value: false });
            }
        }, [props.playbackState]);

        const hideControlAnimation = (value: any) => {
            if (props.isCasting || props.playbackState === 'PAUSED') {
                // When casting, do not hide controls
                return;
            }

            if (hideAnimationTimeoutId.current) {
                clearTimeout(hideAnimationTimeoutId.current);
            }

            hideAnimationTimeoutId.current = setTimeout(() => {
                Animated.parallel([
                    Animated.timing(animations.topControl.opacity, { toValue: 0, delay: 0, useNativeDriver: true }),
                    Animated.timing(animations.topControl.marginTop, { toValue: 0, delay: 0, useNativeDriver: true }),
                    Animated.timing(animations.centerControl.opacity, { toValue: 0, delay: 0, useNativeDriver: true }),
                    Animated.timing(animations.centerControl.marginTop, {
                        toValue: 0,
                        delay: 0,
                        useNativeDriver: true,
                    }),
                    Animated.timing(animations.bottomControl.opacity, { toValue: 0, delay: 0, useNativeDriver: true }),
                    Animated.timing(animations.bottomControl.marginBottom, {
                        toValue: 0,
                        delay: 0,
                        useNativeDriver: true,
                    }),
                ]).start(result => {
                    if (result.finished && isMounted.current) {
                        setState({ name: 'showControls', value: false });
                    }
                });

                if (isMounted.current) {
                    setState({ name: 'animations', value: animations });
                }
            }, value);
        };

        const showControlAnimation = () => {
            Animated.parallel([
                Animated.timing(animations.topControl.opacity, { toValue: 1, useNativeDriver: true }),
                Animated.timing(animations.topControl.marginTop, { toValue: 1, useNativeDriver: true }),
                Animated.timing(animations.centerControl.opacity, { toValue: 1, useNativeDriver: true }),
                Animated.timing(animations.centerControl.marginTop, { toValue: 0, useNativeDriver: true }),
                Animated.timing(animations.bottomControl.opacity, { toValue: 1, useNativeDriver: true }),
                Animated.timing(animations.bottomControl.marginBottom, { toValue: 0, useNativeDriver: true }),
            ]).start(() => {
                hideControlAnimation(playerProps.controlTimeoutDelay);
            });

            if (isMounted.current) {
                setState({ name: 'animations', value: animations });
            }
        };

        const handlePlayPause = () => {
            const newValue = !state.paused;

            setState({ name: 'pauseState', value: newValue });

            if (newValue && props.onPause) {
                props.onPause();
            } else {
                if (props.onPlay) {
                    props.onPlay();
                }
            }
        };

        const handleRewind = () => {
            const value = Math.max(0, props.currentTime - 15000);
            setSeekedTo(value);
            if (props.onRewindPress) {
                props.onRewindPress();
            }
        };

        const handleFastFwd = () => {
            const value = Math.min(props.playbackDuration, props.currentTime + 15000);
            setSeekedTo(value);
            if (props.onForwardPress) {
                props.onForwardPress();
            }
        };

        const handleTracksSelectionEvent = () => {
            setShowTrackSelection(true);
        };

        const handleVolumeEvent = async () => {
            setShowVolumeIcon(false);
        };

        const handleOnScreenTouch = () => {
            if (props.isCasting && state.showControls) {
                return;
            }

            const show: boolean = !state.showControls;
            if (show === true) {
                showControlAnimation();
            } else {
                hideControlAnimation(0);
            }
            setShowVolumeIcon(true);
            setState({ name: 'showControls', value: show });
        };

        const renderProgress = (): JSX.Element => {
            const currentDuration = props.playbackDuration! - props.currentTime!;
            return (
                <View style={[bottomControlStyles.durationContainer]}>
                    <Text style={bottomControlStyles.timerText}>{visualizeVideoDuration(currentDuration!)}</Text>
                </View>
            );
        };

        const renderSeekBar = (): JSX.Element => {
            const hideRealProgress = sliding || props.isLoading;
            const value = hideRealProgress ? seekedTo : props.currentTime;
            const currentTime = visualizeVideoDuration(value);
            const percentComplete = props.playbackDuration! > 0 ? (value / props.playbackDuration!) * 100 : 0;
            const textWidth = 100;
            const textCenter = textWidth / 2;
            const thumbSize = 25;
            const thumbOffset = (percentComplete * thumbSize) / 100 - thumbSize / 2;
            const left = Math.max((percentComplete * layout.width) / 100, 0) - (textCenter + thumbOffset);

            return (
                <View style={[bottomControlStyles.sliderContainer]}>
                    <View style={bottomControlStyles.sliderWrapper}>
                        <Slider
                            onLayout={onLayout}
                            maximumValue={props.playbackDuration}
                            minimumValue={0}
                            step={1}
                            value={value}
                            style={bottomControlStyles.slider}
                            onSlidingStart={() => {
                                setSliding(true);
                            }}
                            onSlidingComplete={(value: number) => {
                                setSliding(false);
                                hideControlAnimation(playerProps.controlTimeoutDelay);
                                if (props.onSlidingComplete) {
                                    props.onSlidingComplete(value);
                                }
                            }}
                            onValueChange={(value: number) => {
                                setSeekedTo(value);
                                setState({ name: 'pauseState', value: false });
                                hideControlAnimation(playerProps.controlTimeoutDelay);
                            }}
                            thumbTintColor={colors.brandTint}
                            minimumTrackTintColor={colors.brandTint}
                            maximumTrackTintColor={'rgba(256, 256, 256, 0.2)'}
                        />
                        <Text style={[bottomControlStyles.currentTime, { left: left, width: textWidth }]}>
                            {currentTime}
                        </Text>
                    </View>
                    {renderProgress()}
                </View>
            );
        };
        const renderBottomControls = () => {
            const seekBarControl = renderSeekBar();
            const value = seekVolumeTo;
            const volumeIcon = seekVolumeTo > 0 ? <UnMuteIcon /> : <MuteIcon />;

            const mainResource = props.resource;
            let name = mainResource && mainResource.name;
            if (mainResource && mainResource.type === 'tvepisode') {
                name = `${mainResource.name}: S${mainResource.seasonNumber} E${mainResource.episodeNumber}`;
            }
            return (
                <Animated.View
                    useNativeDriver={true}
                    style={[
                        bottomControlStyles.bottom,
                        {
                            opacity: state.animations.bottomControl.opacity,
                            translateY: state.animations.bottomControl.marginBottom,
                        },
                    ]}>
                    <View style={bottomControlStyles.titleContainer}>
                        <View>
                            {props.isCasting && <Text style={bottomControlStyles.titleText}>{props.castLabel}</Text>}
                            {props.resource && <Text style={bottomControlStyles.titleText}>{name}</Text>}
                            {props.resource && (
                                <Text style={bottomControlStyles.captionText}>{props.resource.providerName}</Text>
                            )}
                            {props.resource && (
                                <Text style={[bottomControlStyles.captionText]}>{props.resource.releaseYear}</Text>
                            )}
                        </View>
                        <View style={bottomControlStyles.volumeContainer}>
                            {!showVolumeIcon && (
                                <Slider
                                    onLayout={onLayout}
                                    maximumValue={100}
                                    minimumValue={0}
                                    step={1}
                                    value={value}
                                    style={bottomControlStyles.volumeSlider}
                                    onSlidingComplete={(value: number) => {
                                        console.debug('Volume onSlideComplete ', value);
                                        hideControlAnimation(playerProps.controlTimeoutDelay);
                                        setShowVolumeIcon(true);
                                    }}
                                    onValueChange={(value: number) => {
                                        setSeekVolumeTo(value);
                                        SystemSetting.setVolume(value / 100);
                                        hideControlAnimation(playerProps.controlTimeoutDelay);
                                        props.handleVolumeToggle && props.handleVolumeToggle(value / 100);
                                    }}
                                    thumbTintColor={colors.primary}
                                    minimumTrackTintColor={colors.primary}
                                    maximumTrackTintColor={'rgba(256, 256, 256, 0.2)'}
                                />
                            )}
                            {showVolumeIcon &&
                                renderControlIcon(
                                    'VolumeButton',
                                    volumeIcon,
                                    handleVolumeEvent,
                                    bottomControlStyles.icon,
                                )}
                        </View>
                    </View>
                    {seekBarControl}
                </Animated.View>
            );
        };

        const renderControlIcon = (
            accessibilityLabel: string,
            children: any,
            callback: any,
            style = {},
        ): JSX.Element => {
            return (
                <TouchableHighlight
                    accessibilityLabel={accessibilityLabel}
                    underlayColor={colors.backgroundInactiveSelected}
                    onPress={() => {
                        if (callback) {
                            //  hideControlAnimation(playerProps.controlTimeoutDelay);
                            callback();
                        }
                    }}
                    style={[style]}>
                    {children}
                </TouchableHighlight>
            );
        };

        // const renderCastButton = () => {
        //     if (props.showCastIcon && !Platform.isTV) {
        //         const { CastButton } = require('react-native-google-cast');
        //         return (
        //             <CastButton
        //                 style={[
        //                     props.isCasting && props.castType === 'Chromecast'
        //                         ? topControlStyles.activeCastIcon
        //                         : topControlStyles.castIconContainer,
        //                 ]}
        //             />
        //         );
        //     }
        // };

        const renderTopControls = () => {
            return (
                <Animated.View
                    style={[
                        topControlStyles.top,
                        {
                            opacity: state.animations.topControl.opacity,
                        },
                    ]}>
                    <View style={topControlStyles.rootContainer}>
                        <View style={{ flexDirection: 'row' }}>
                            {renderControlIcon(
                                'TracksButton',
                                <TracksIcon />,
                                handleTracksSelectionEvent,
                                topControlStyles.moreControlsIconContainer,
                            )}
                            {Platform.OS === 'ios' && !Platform.isTV && !props.isDownloadedContentPlayback && (
                                <QpNxgAirplayView style={topControlStyles.castIconContainer} />
                            )}
                            {/* {renderCastButton()} */}
                        </View>
                        <View>
                            {renderControlIcon('CloseButton', <CloseIcon />, props.onBackPress, [
                                topControlStyles.moreControlsIconContainer,
                            ])}
                        </View>
                    </View>
                </Animated.View>
            );
        };

        const renderCenterControls = () => {
            const PlayStateIcon = state.paused === true ? PlayIcon : PauseIcon;
            return (
                <Animated.View
                    style={[centerControlStyles.center, { opacity: state.animations.centerControl.opacity }]}>
                    <View style={{ flexDirection: 'row' }}>
                        {renderControlIcon('RewindButton', <RewindIcon />, handleRewind, {
                            marginRight: '10%',
                            ...centerControlStyles.icon,
                        })}
                        {renderControlIcon('Play', <PlayStateIcon />, handlePlayPause, {
                            width: 50,
                            height: 50,
                            justifyContent: 'center',
                            alignItems: 'center',
                            borderRadius: 30,
                        })}
                        {renderControlIcon('ForwardButton', <ForwardIcon />, handleFastFwd, {
                            marginLeft: '10%',
                            ...centerControlStyles.icon,
                        })}
                    </View>
                </Animated.View>
            );
        };
        return (
            <TouchableHighlight
                accessibilityLabel={'PlayerControls'}
                underlayColor={'transparent'}
                onPress={handleOnScreenTouch}
                style={topControlStyles.container}>
                <>
                    <View style={topControlStyles.container}>
                        {state.showControls && (
                            <View style={topControlStyles.container}>
                                <View style={{ flex: 1 }}>{renderTopControls()}</View>
                                <View style={{ flex: 1 }}>{renderCenterControls()}</View>
                                <View style={{ flex: 1 }}>{renderBottomControls()}</View>
                            </View>
                        )}
                        {showTrackSelection && (
                            <TrackSelectionView
                                captionOptions={props.captionOptions!}
                                audioOptions={props.audioOptions!}
                                activeTextTrack={props.activeTextTrack}
                                activeAudioTrack={props.activeAudioTrack}
                                onCancel={() => setShowTrackSelection(false)}
                                onCaptionOptionSelected={props.onTextOptionSelected!}
                                onAudioOptionSelected={props.onAudioOptionSelected!}
                            />
                        )}
                    </View>
                </>
            </TouchableHighlight>
        );
    },
);
