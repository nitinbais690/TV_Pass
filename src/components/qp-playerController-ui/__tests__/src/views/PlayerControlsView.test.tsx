import React from 'react';
import { PlayerControlsView } from '../../../src/views/PlayerControlsView';
import { render, toJSON, getByLabelText, fireEvent } from '@testing-library/react-native';

describe('PlayerControlsView', () => {
    it('should render loading', () => {
        jest.useFakeTimers();
        const jsx = <PlayerControlsView isLoading={true} />;
        const { container } = render(jsx);
        expect(toJSON(container)).toMatchSnapshot();
    });

    it('should be visible on pressing', () => {
        jest.useFakeTimers();
        const jsx = <PlayerControlsView />;
        const { container, debug } = render(jsx);
        const playerControls = getByLabelText(container, 'PlayerControls');
        fireEvent.press(playerControls);
        debug();
        expect(getByLabelText(container, 'CloseButton')).toBeDefined();
        expect(getByLabelText(container, 'RewindButton')).toBeDefined();
        expect(getByLabelText(container, 'ForwardButton')).toBeDefined();
        expect(getByLabelText(container, 'Play')).toBeDefined();
    });

    it('should render all player controls', () => {
        jest.useFakeTimers();
        const jsx = <PlayerControlsView showOnStart={true} currentTime={207360000} playbackDuration={307360000} />;

        const { container } = render(jsx);
        expect(toJSON(container)).toMatchSnapshot();
        expect(getByLabelText(container, 'CloseButton')).toBeDefined();
        expect(getByLabelText(container, 'RewindButton')).toBeDefined();
        expect(getByLabelText(container, 'ForwardButton')).toBeDefined();
        expect(getByLabelText(container, 'Play')).toBeDefined();
    });

    it('should call player controls on press', () => {
        jest.useFakeTimers();
        const pausePressHandler = jest.fn();
        const rewindPressHandler = jest.fn();
        const forwardPressHandler = jest.fn();
        const backPressHandler = jest.fn();

        const jsx = (
            <PlayerControlsView
                showOnStart={true}
                onPause={pausePressHandler}
                onRewindPress={rewindPressHandler}
                onForwardPress={forwardPressHandler}
                onBackPress={backPressHandler}
                isLoading={false}
            />
        );

        const { container } = render(jsx);

        const playButton = getByLabelText(container, 'Play');
        fireEvent.press(playButton);
        expect(pausePressHandler).toHaveBeenCalled();

        const rewindButton = getByLabelText(container, 'RewindButton');
        fireEvent.press(rewindButton);
        expect(rewindPressHandler).toHaveBeenCalled();

        const forwardButton = getByLabelText(container, 'ForwardButton');
        fireEvent.press(forwardButton);
        expect(forwardPressHandler).toHaveBeenCalled();

        const closeButton = getByLabelText(container, 'CloseButton');
        fireEvent.press(closeButton);
        expect(backPressHandler).toHaveBeenCalled();
    });
});
