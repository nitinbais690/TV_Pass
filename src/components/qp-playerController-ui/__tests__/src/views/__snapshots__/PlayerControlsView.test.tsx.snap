// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerControlsView should render all player controls 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="PlayerControls"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgba(0,0,0,0.7)",
                "flex": 1,
                "flexDirection": "row",
                "opacity": 1,
                "paddingHorizontal": 30,
                "paddingTop": 10,
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                }
              }
            >
              <View
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  accessibilityLabel="TracksButton"
                  accessible={true}
                  focusable={true}
                  hasTVPreferredFocus={false}
                  isTVSelectable={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "height": 40,
                        "justifyContent": "center",
                        "width": 40,
                      },
                    ]
                  }
                >
                  <SvgMock />
                </View>
                <View
                  style={
                    Object {
                      "flexGrow": 1,
                      "height": 40,
                      "marginLeft": 25,
                      "tintColor": "#ffffff",
                      "width": 40,
                    }
                  }
                />
              </View>
              <View>
                <View
                  accessibilityLabel="CloseButton"
                  accessible={true}
                  focusable={true}
                  hasTVPreferredFocus={false}
                  isTVSelectable={true}
                  style={
                    Array [
                      Array [
                        Object {
                          "alignItems": "center",
                          "borderRadius": 20,
                          "height": 40,
                          "justifyContent": "center",
                          "width": 40,
                        },
                      ],
                    ]
                  }
                >
                  <SvgMock />
                </View>
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 2,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(0,0,0,0.7)",
                "flex": 2,
                "justifyContent": "center",
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Object {
                  "flexDirection": "row",
                }
              }
            >
              <View
                accessibilityLabel="RewindButton"
                accessible={true}
                focusable={true}
                hasTVPreferredFocus={false}
                isTVSelectable={true}
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "marginRight": "10%",
                    },
                  ]
                }
              >
                <SvgMock />
              </View>
              <View
                accessibilityLabel="Play"
                accessible={true}
                focusable={true}
                hasTVPreferredFocus={false}
                isTVSelectable={true}
                style={
                  Array [
                    Array [
                      Object {
                        "borderRadius": 50,
                      },
                      Object {},
                    ],
                  ]
                }
              >
                <SvgMock />
              </View>
              <View
                accessibilityLabel="ForwardButton"
                accessible={true}
                focusable={true}
                hasTVPreferredFocus={false}
                isTVSelectable={true}
                style={
                  Array [
                    Object {
                      "borderRadius": 50,
                      "marginLeft": "10%",
                    },
                  ]
                }
              >
                <SvgMock />
              </View>
            </View>
          </View>
        </View>
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "stretch",
                "backgroundColor": "rgba(0,0,0,0.7)",
                "flex": 1,
                "justifyContent": "flex-end",
                "opacity": 1,
                "paddingBottom": 10,
                "paddingHorizontal": 50,
                "translateY": 0,
              }
            }
            useNativeDriver={true}
          >
            <View
              style={
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "paddingBottom": 15,
                }
              }
            >
              <View />
              <View
                style={
                  Object {
                    "alignContent": "center",
                    "flexDirection": "row",
                    "paddingBottom": 25,
                  }
                }
              >
                <View
                  accessibilityLabel="VolumeButton"
                  accessible={true}
                  focusable={true}
                  hasTVPreferredFocus={false}
                  isTVSelectable={true}
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "borderRadius": 20,
                        "height": 40,
                        "justifyContent": "center",
                        "width": 40,
                      },
                    ]
                  }
                >
                  <SvgMock />
                </View>
              </View>
            </View>
            <View
              style={
                Array [
                  Object {
                    "alignItems": "flex-start",
                    "flexDirection": "row",
                    "justifyContent": "flex-start",
                  },
                ]
              }
            >
              <View
                style={
                  Object {
                    "flex": 1,
                    "height": 45,
                    "paddingBottom": 30,
                  }
                }
              >
                <RNCSlider
                  disabled={false}
                  enabled={true}
                  maximumTrackTintColor="rgba(256, 256, 256, 0.2)"
                  maximumValue={307360000}
                  minimumTrackTintColor="#686EFF"
                  minimumValue={0}
                  step={1}
                  style={
                    Array [
                      Object {
                        "height": 40,
                      },
                      Object {
                        "alignItems": "center",
                        "flex": 1,
                        "flexDirection": "column",
                        "height": 25,
                      },
                    ]
                  }
                  thumbTintColor="#686EFF"
                  value={207360000}
                />
                <Text
                  style={
                    Array [
                      Object {
                        "bottom": 0,
                        "color": "white",
                        "fontFamily": "Inter-Regular",
                        "fontSize": 14.742857142857144,
                        "position": "absolute",
                        "textAlign": "center",
                      },
                      Object {
                        "left": -54.36621551275377,
                        "width": 100,
                      },
                    ]
                  }
                >
                  57:36:00
                </Text>
              </View>
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "flex-start",
                      "flexDirection": "row",
                      "marginLeft": 8,
                      "paddingTop": 0,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#ffffff",
                      "fontSize": 14.742857142857144,
                      "paddingLeft": 7.5,
                      "textAlign": "right",
                    }
                  }
                >
                  27:46:40
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`PlayerControlsView should render loading 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="PlayerControls"
    accessible={true}
    focusable={true}
    hasTVPreferredFocus={false}
    isTVSelectable={true}
    style={
      Object {
        "flex": 1,
        "width": "100%",
      }
    }
  >
    <View
      collapsable={false}
      style={
        Object {
          "flex": 1,
          "width": "100%",
        }
      }
    />
  </View>
</View>
`;
